lib/models/obms.js

```diff
@@ -0,0 +1,183 @@
+// Copyrighr 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = ObmsModelFactory;
+
+ObmsModelFactory.$provide = 'Models.Obms';
+ObmsModelFactory.$inject = [
+    'Model',
+    'Services.Waterline',
+    '_',
+    'Services.Encryption',
+    'Constants',
+    'Promise',
+    'util',
+    'Errors',
+    'Assert'
+];
+
+function ObmsModelFactory (
+    Model,
+    waterline,
+    _,
+    encryption,
+    Constants,
+    Promise,
+    util,
+    Errors,
+    assert
+) {
+    var secretsPattern = _.reduce(Constants.Logging.Redactions, function(pattern, regex) {
```

Comment:
@yyscamper Actually, I think it would be even better if we a custom validation called `secret` at the model level.  That would allow us to make secrets management totally generic within the models.  If a model has a password property, rather than a regex match we'd defiine it like so in the model:

```
password: {
      type: 'string',
      secret: true,
}
```